Src/ResistanceOperationOverhaul/Classes/ResistanceOverhaulHelpers.uc

```diff
@@ -97,11 +96,13 @@ private static function string RandomlyChooseCharacter(name ClassName, array<str
 		`LOG("RandomlyChooseCharacter checking " $ Character.GetFullName());
 		if (DisallowedCharacters.Find(Character.GetFullName()) == INDEX_NONE)
 		{
-			`LOG("RandomlyChooseCharacter not disallowed");
 			if (CharacterIsValid(Character, ClassTemplate))
 			{
-				`LOG("RandomlyChooseCharacter adding to RandomCharacterOptions");
-				RandomCharacterOptions.AddItem(Character.GetFullName());
+				`LOG("Searching for Character with Class: " $ ClassName);
+				if(ClassName == Character.GetSoldierClassTemplateName()) {
```

Comment:
1. Curly brackets should be on new lines
2. I don't like the side-effect where, if there aren't any characters with the given class, then the soldier will end up being fully randomly generated. I think that if it can't find a character with the given class, it should just randomly choose a character. And then if it can't find a character, it should be randomly generated.